### PR TITLE
Add authentication for selected endpoints

### PR DESCRIPTION
**Pull request by jav1palace**

### CODE REVIEW

**Review by Zohane87 — COMMENTED** [review]
Let's have a chat

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    const user = await this.userRepository.findOneBy({
+      username,
+    });
+
+    const canLogin =
+      user?.isActive &&
+      (await this.encryptionService.isPasswordCorrect(pass, user.password));
+
+    if (canLogin) {
+      const { password, ...result } = user;
```
Comment (jav1palace) [inline src/auth/auth.service.ts:30]:
```suggestion
      const { ...result } = user;
```

File: src/database/database.module.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { Module } from '@nestjs/common';
 import { TypeOrmModule } from '@nestjs/typeorm';
+
```
[the remark this answers is not in the document]
Comment (jav1palace) [reply src/database/database.module.ts:3]: Typescript Hero plugin leaves it like this.

File: src/auth/auth.service.spec.ts [hunk 3]
```diff
@@ -41,28 +44,25 @@ describe('LoginService', () => {
...
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
 
   it('login', () => {
-    expect(
-      service.login({ username: 'username', password: 'password' }),
-    ).resolves.toBe(true);
+    const { password, ...response } = user;
```
Comment (jav1palace) [inline src/auth/auth.service.spec.ts:55]:
```suggestion
    const { ...response } = user;
```

File: src/auth/auth.service.spec.ts [hunk 4]
```diff
@@ -1,12 +1,14 @@
 import { BadRequestException } from '@nestjs/common';
```
Comment (jav1palace) [inline src/auth/auth.service.spec.ts:1]:
```suggestion
```

File: src/auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      return result;
```
[the remark this answers is not in the document]
Comment (jav1palace) [reply src/auth/auth.service.ts:31]: I don't like it either but It was really hard to read with a single one.

**Review by Zohane87 — APPROVED** [review]
Reviewed! :)

**Pull request merged**